### PR TITLE
Add eval recipe

### PR DESCRIPTION
**Pull request by Louis-Dupont**
This PR depends on https://github.com/Deci-AI/super-gradients/pull/386

### CODE REVIEW

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 1]
```diff
@@ -221,6 +214,73 @@ def train_from_config(cls, cfg: Union[DictConfig, dict]) -> None:
                       valid_loader=val_dataloader,
                       training_params=cfg.training_hyperparams)
 
+    @classmethod
+    def resume_experiment(cls, experiment_name: str, ckpt_root_dir: str = None) -> None:
```
Comment (shaydeci) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:218]: Maybe add some example how it can and should be used.

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 2]
```diff
@@ -221,6 +214,73 @@ def train_from_config(cls, cfg: Union[DictConfig, dict]) -> None:
...
+        :param experiment_name:     Name of the experiment to resume
+        :param ckpt_root_dir:       Directory including the checkpoints
+        """
+        logger.info("Resume training using the checkpoint recipe, ignoring the current recipe")
+        cfg = load_experiment_cfg(experiment_name, ckpt_root_dir)
+        add_params_to_cfg(cfg, params=["training_hyperparams.resume=True"])
+        cls.train_from_config(cfg)
+
+    @classmethod
+    def test_from_recipe(cls, cfg: DictConfig) -> None:
```
Comment (shaydeci) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:231]: Maybe add some example how it can and should be used.

File: src/super_gradients/training/utils/checkpoint_utils.py [hunk 3]
```diff
@@ -1,15 +1,63 @@
...
 try:
     from torch.hub import download_url_to_file, load_state_dict_from_url
 except (ModuleNotFoundError, ImportError, NameError):
     from torch.hub import _download_url_to_file as download_url_to_file
 
 
+logger = get_logger(__name__)
+
+
+def load_experiment_cfg(experiment_name: str, ckpt_root_dir: str = None) -> DictConfig:
```
Comment (shaydeci) [inline src/super_gradients/training/utils/checkpoint_utils.py:24]: Add docs.

**Review by shaydeci — CHANGES_REQUESTED** [review]
Nice work!
I think there's a lot to explain here and the docs should be more detailed, we have a great feature here and it'd be a shame to not have it well documented.

File: src/super_gradients/examples/test_from_recipe_example/test_from_recipe.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""
+Example code for testing SuperGradient's recipes.
+NOTE:   This script does NOT run TRAINING,
+        so make sure in the recipe that you load a PRETRAINED MODEL
+        either from one of your checkpoint or from a pretrained model.
+
+General use: python test_from_recipe.py --config-name="DESIRED_RECIPE".
+For recipe's specific instructions and details refer to the recipe's configuration file in the recipes directory.
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/examples/test_from_recipe_example/test_from_recipe.py]: updated

File: src/super_gradients/examples/test_experiment_example/test_experiment.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+"""
+Example code for testing the output model of an experiment.
+
+General use: python test_experiment.py --experiment_name=<PREVIOUSLY-RUN-EXPERIMENT>
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/examples/test_experiment_example/test_experiment.py]: updated

File: src/super_gradients/examples/resume_experiment_example/resume_experiment.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+"""
+Example code for resuming SuperGradient's recipes.
+
+General use: python resume_experiment.py --experiment_name=<PREVIOUSLY-RUN-EXPERIMENT>
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/examples/resume_experiment_example/resume_experiment.py:4]: updated

File: src/super_gradients/training/sg_trainer/sg_trainer.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:220]: updated

File: src/super_gradients/training/utils/checkpoint_utils.py (hunk 3 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/utils/checkpoint_utils.py]: updated

**Review by shaydeci — APPROVED** [review]
LGTM
